src/js/index.js

```diff
@@ -1 +1,44 @@
 import "../styles/main.scss";
+
+const $ = (selector) => document.querySelector(selector)
+
+const emailRegex = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+
+$(".js-email-input").addEventListener("keyup", (e) => {
+    let email = e.target.value
+    if (!emailRegex.test(email)) {
+        $(".js-error").textContent = "Please enter a valid email"
+        $(".js-submit-button").disabled = true
+    }else {
+        $(".js-error").textContent = ""
+        $(".js-submit-button").disabled = false
+    }
+})
+$(".js-form").addEventListener("submit", (e) => e.preventDefault())
+
+$(".js-submit-button").addEventListener("click", () => {
+    let email = $(".js-email-input").value
+    if (!emailRegex.test(email)) {
+        $(".js-error").textContent = "Please enter a valid email"
+        $(".js-submit-button").disabled = true
+    }else {
```

Comment:
instead of the `else` just breakout early with a `return`

Reply:
Also here: . (class) or # (id)? I would think this would all be specific (# = id) not general (class)? Or am I missing something (or just not up-to-date on "proper" conventions)?

Reply:
There are no conventions. Id selection is faster and more specific. By using IDs we also don't need to add js- prefixes to IDs. So if we can use ids then great.